CodeAssignment_MVVM_Combine/Sources/ViewModels/DetailView/DetailViewModel.swift

```diff
@@ -0,0 +1,58 @@
+//
+//  DetailViewModel.swift
+//  CodeAssignment_MVVM_Combine
+//
+//  Created by gyusoku.i on 2024/05/22.
+//
+
+import Foundation
+import UIKit
+import Combine
+
+final class DetailViewModel {
+    
+    private let apiClient = APIClient()
+    /** 星付けの状態
+    - PassthroughtSubjectからPublishedに変更
+    - private(set)に設定し、クラスの外部では読み取りだけを可能にする
+    - Subjectじゃなく、Publishedを使うと、sendメソッドを使わずに受け取る側(VC)で値の変更を検知可能
+     */
+    @Published private(set) var isStarred: Bool = false
+    /// 詳細画面で扱うリポジトリ
+    @Published private(set) var repository: Repositories.Repository
+
+    /** initializerでリポジトリを持っておく
+    - ViewModelにリポジトリデータを持っておき、データの加工をViewModel側で処理するようにする
+     */
+    init(repository: Repositories.Repository) {
+        self.repository = repository
+    }
+
+    /** POST・DELETE リクエストを送信し、repositoryにスターの付け・解除するメソッド
+    - ViewControllerのUIから持ってくるのではなく、モデルのデータを利用するようにする
+     */
+    func changeRepositoryStarStatus() {
+        let requestProtocol = GitHubStarRepositoryRequest(owner: repository.owner.userName, repository: repository.name, starStatus: !isStarred)
```

Comment:
好みの問題だけど、引数が長くなる時は改行してあげたほうが見やすい

```suggestion
        let requestProtocol = GitHubStarRepositoryRequest(
            owner: repository.owner.userName,
            repository: repository.name,
            starStatus: !isStarred
        )
```

Reply:
加えて、`GitHubStarRepositoryRequest` はプロトコルではないので変数名が不自然ですね。
普通に `let request` とかでいいと思う

Reply:
[cd05ef6](https://github.com/KyusokLee/CodeAssignment_MVVM_Combine/pull/6/commits/cd05ef6cbd4c6332eb7e536caa2308f7d44946e7)にてコード修正しました！ご確認お願いします！